src/libostree/ostree-repo-refs.c

```diff
@@ -107,7 +107,38 @@ write_checksum_file_at (OstreeRepo   *self,
 
     if (!_ostree_repo_file_replace_contents (self, dfd, name, (guint8*)bufnl, l + 1,
                                              cancellable, error))
-      goto out;
+      {
+        if (g_error_matches (*error, G_IO_ERROR, G_IO_ERROR_IS_DIRECTORY))
+          {
+            g_autoptr(GHashTable) refs = NULL;
+            GHashTableIter hashiter;
+            gpointer hashkey, hashvalue;
+
+            g_clear_error (error);
+
+            if (!ostree_repo_list_refs (self, name, &refs, cancellable, error))
+              goto out;
+
+            g_hash_table_iter_init (&hashiter, refs);
+
+            while ((g_hash_table_iter_next (&hashiter, &hashkey, &hashvalue)))
+              {
+                if (strcmp (name, (char *)hashkey) != 0)
```

Comment:
Hm, so how this works is, let's say our name is "foo/bar". When list_refs grabs all refs with prefix foo/bar, it will put into refs "foo/bar/..." AND "foo/bar" if it exists. It's not an error when we specify "foo/bar" to overwrite, but it would be if there is a "foo/bar/baz". That check basically says if it's not exactly "foo/bar", then we don't like it.